observation/interferometric_pointing.py

```diff
@@ -57,29 +62,23 @@ class NoTargetsUpError(Exception):
 
         start_time = time.time()
         targets_observed = []
-        # Keep going until the time is up
-        keep_going = True
-        while keep_going:
-            keep_going = (opts.max_duration is not None)
-            targets_before_loop = len(targets_observed)
-            # Iterate through source list, picking the next one that is up
-            for target in observation_sources.iterfilter(el_limit_deg=opts.horizon):
-                # Check if all offset pointings in compound scan will be up
-                compound_steps = 1 + 2 * (opts.number_of_steps // 2)
-                # Add some extra time for slews between pointings
-                step_duration = opts.track_duration + 4.
-                compound_duration = compound_steps * step_duration
-                if not session.target_visible(target, duration=compound_duration):
-                    continue
-                session.label('interferometric_pointing')
-                session.track(target, duration=opts.track_duration, announce=False)
-                for direction in {'x', 'y'}:
-                    for offset in np.linspace(-opts.max_extent, opts.max_extent,
-                                              opts.number_of_steps // 2):
-                        if direction == 'x':
-                            offset_target = [offset, 0.0]
-                        else:
-                            offset_target = [0.0, offset]
+        try: # Keep going until the time is up or interrupted
+            keep_going = True
+            while keep_going:
+                keep_going = (opts.max_duration is not None)
+                targets_before_loop = len(targets_observed)
+                # Iterate through source list, picking the next one that is up
+                for target in observation_sources.iterfilter(el_limit_deg=opts.horizon):
+                    # Check if all offset pointings in compound scan will be up
+                    compound_steps = opts.number_of_steps
```

Comment:
This seems like an unnecessary variable now.

Reply:
I agree - it's now removed in most recent commit